src/list.ts

```diff
@@ -0,0 +1,259 @@
+const editDescriptionButton = document.getElementById('editDescriptionButton') as HTMLButtonElement;
+
+const shoppinglist = database.getDocument(
+    CONFIG.DATABASE_ID,
+    CONFIG.DB_COLLECTION_SHOPPINGLISTS,
+    listid
+) as Promise<Einkaufsliste>;
+
+shoppinglist.then((list) => {
+    realtimeUpdate(list);
+
+    updateListInfo(list);
+
+    editDescriptionButton.addEventListener('click', () => {
+        if (!listDescription.disabled.valueOf()) {
+            database.updateDocument(
+                CONFIG.DATABASE_ID,
+                CONFIG.DB_COLLECTION_SHOPPINGLISTS,
+                list.$id!,
+                {
+                    stichtag: deadline.value,
+                    beschriftung: listTitel.value,
+                    beschreibung: listDescription.value
+                }
+            );
+        }
+        listTitel.disabled = !listTitel.disabled.valueOf();
+        listDescription.disabled = !listDescription.disabled.valueOf();
+        deadline.disabled = !deadline.disabled.valueOf();
+    });
+
+    form.addEventListener('submit', (event) => {
+        event.preventDefault();
+
+        const product: Listeneintrag = {
+            artikelname: productName.value,
+            anzahl: Number.parseInt(quantity.value),
+            erledigt: false,
+            einkaufslisten: list.$id!
+        }
+
+        database.createDocument(
+            CONFIG.DATABASE_ID,
+            CONFIG.DB_COLLECTION_SHOPLISTENTRY,
+            ID.unique(),
+            product,
+            [
+                Permission.read(Role.team(list.ID_Household)),
+                Permission.update(Role.team(list.ID_Household)),
+                Permission.delete(Role.team(list.ID_Household)),
+            ],
+        ).then(() => { },
+            () => { window.alert('Das geforderte Produkt konnte nicht erstellt werden. :(') });
+    });
+
+    list.listeneintrag.forEach(function (value: Listeneintrag) {
+        productTable.appendChild(createTableRow(value));
+    });
+}, () => {
+    window.location.pathname = '/uebersicht.html';
+});
+
+function createTableRow(data: Listeneintrag, id?: string): HTMLTableRowElement {
+    const row = document.createElement('tr');
+    if (id) {
+        row.setAttribute('id', id!);
+    } else {
+        row.setAttribute('id', data.$id!);
+    }
+
+    const checkbox = document.createElement('input');
+    checkbox.type = 'checkbox';
+    checkbox.checked = data.erledigt;
+    checkbox.addEventListener('change', async () => {
+        data.erledigt = Boolean(checkbox.checked);
+        await updateListEntry(data);
+    });
+
+    const productName = document.createElement('input');
+    productName.type = 'text';
+    productName.disabled = true;
+    productName.value = data.artikelname;
+
+    const quantity = document.createElement('input');
+    quantity.type = 'number';
+    quantity.min = '1';
+    quantity.max = '999';
+    quantity.value = String(data.anzahl);
```

Comment:
conversion number to string is handled automatically, iirc. the explicit cast might not be necessary.